receiver/dotnetdiagnosticsreceiver/dotnet/request.go

```diff
@@ -0,0 +1,240 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnet
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/network"
+)
+
+// RequestWriter submits the initial request to the dotnet diagnostics backend,
+// after which it starts sending us metrics at the specified interval.
+// ByteBuffer is swappable for testing.
+type RequestWriter struct {
+	buf         network.ByteBuffer
+	w           io.Writer
+	intervalSec int
+	names       []string
+}
+
+func NewRequestWriter(buf network.ByteBuffer, w io.Writer, intervalSec int, names ...string) *RequestWriter {
+	return &RequestWriter{buf: buf, w: w, intervalSec: intervalSec, names: names}
+}
+
+func (w *RequestWriter) SendRequest() error {
+	req, err := w.createRequest()
+	if err != nil {
+		return err
+	}
+	_, err = w.w.Write(req)
+	return err
+}
+
+const eventPipeCommand = 2
+const collectTracing2CommandID = 3
```

Comment:
Where do this const values come from? Is this some dotnet thing? Can you add a link to doc if it exists?

Reply:
Yeah, it's a dotnet thing -- I got these values from Microsoft's codebase. I have added links to MS's source and docs.

Also, I had these constants in their own file, `constants.go` and I think that helped provide context, but since a lot of the constants weren't used, the file failed lint checking. Not sure what the best approach is, but here's what it looked like before I removed it for the linter, and extracted just the constants that were actually in use (more of these are in use in the full PR):

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2200/files#diff-79992f841b9c789e52a5d3eca0146372213595b358a1bad1e05e7a7b605dede7